packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -56,6 +56,28 @@ public ModelProvider(InputModelType inputModel)
             _isStruct = inputModel.ModelAsStruct;
         }
 
+        public bool IsUnknownDiscriminatorModel => _inputModel.IsUnknownDiscriminatorModel;
+
+        public string? DiscriminatorValue => _inputModel.DiscriminatorValue;
+
+        private IReadOnlyList<ModelProvider>? _derivedModels;
+        public IReadOnlyList<ModelProvider> DerivedModels => _derivedModels ??= BuildDerivedModels();
```

Comment:
So I think this `DerivedModels` is not really the derived models based on its implementation.
It is only those `Discriminated Children` types, could we rename it so that we will not misunderstand the purpose of this property?